flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/ttl/RocksDbTtlCompactFiltersManager.java

```diff
@@ -108,7 +110,16 @@ public void configCompactFilter(
 			@Nonnull StateDescriptor<?, ?> stateDesc,
 			TypeSerializer<?> stateSerializer) {
 		StateTtlConfig ttlConfig = stateDesc.getTtlConfig();
-		if (ttlConfig.isEnabled() && ttlConfig.getCleanupStrategies().inRocksdbCompactFilter()) {
+
+		//if ttl is enable and cleanup in background and user did not specify the cleanup strategy
+		//then build the compact filter cleanup strategy and apply the default value
+		if (ttlConfig.isEnabled() && ttlConfig.isCleanupInBackground() &&
+			ttlConfig.getCleanupStrategies().getRocksdbCompactFilterCleanupStrategy() == null) {
+			ttlConfig.getCleanupStrategies().activate(ROCKSDB_COMPACTION_FILTER,
+				new StateTtlConfig.RocksdbCompactFilterCleanupStrategy(1000L));
+		}
+
+		if (ttlConfig.getCleanupStrategies().inRocksdbCompactFilter()) {
```

Comment:
Maybe we need to add `ttlConfig.isEnabled()` here?

Reply:
Oh... you are right, I forgot.